pennylane/spin/spin_hamiltonian.py

```diff
@@ -298,3 +309,347 @@ def fermi_hubbard(
+            a square matrix of size ``(num_spins, num_spins)``, where ``num_spins`` is the total
+            number of spins. Default value is 1.0.
+        coulomb (float or List[float]): Coulomb interaction between spins. It can be a constant or a
+            list of length equal to number of spins.
+        intersite_int (float or List[float] or List[math.array(float)]): Interaction strength between spins on
+            neighbouring sites, it can be a number, a list of length equal to ``neighbour_order`` or
+            a square matrix of size ``(num_spins, num_spins)``, where ``num_spins`` is the total
+            number of spins. Default value is 1.0.
+        boundary_condition (bool or list[bool]): Defines boundary conditions for different lattice
+            axes, default is ``False`` indicating open boundary condition.
+        neighbour_order (int): Specifies the interaction level for neighbors within the lattice.
+            Default is 1, indicating nearest neighbours.
+        mapping (str): Specifies the fermion-to-qubit mapping. Input values can be
+            ``'jordan_wigner'``, ``'parity'`` or ``'bravyi_kitaev'``.
+
+    Returns:
+       ~ops.op_math.Sum: Hamiltonian for the Emery model.
+
+    **Example**
+
+    >>> n_cells = [2]
+    >>> h = [0.5]
+    >>> u = 1.0
+    >>> v = 0.2
+    >>> spin_ham = qml.spin.emery("chain", n_cells, hopping=h, coulomb=u, intersite_int=v)
+    >>> spin_ham
+    (
+    -0.25 * (Y(0) @ Z(1) @ Y(2))
+    + -0.25 * (X(0) @ Z(1) @ X(2))
+    + 0.7000000000000002 * I(0)
+    + -0.25 * (Y(1) @ Z(2) @ Y(3))
+    + -0.25 * (X(1) @ Z(2) @ X(3))
+    + -0.35 * Z(1)
+    + -0.35 * Z(0)
+    + 0.25 * (Z(0) @ Z(1))
+    + -0.35 * Z(3)
+    + -0.35 * Z(2)
+    + 0.25 * (Z(2) @ Z(3))
+    + 0.05 * (Z(0) @ Z(2))
+    + 0.05 * (Z(0) @ Z(3))
+    + 0.05 * (Z(1) @ Z(2))
+    + 0.05 * (Z(1) @ Z(3))
+    )
+
+    """
+
+    lattice = _generate_lattice(lattice, n_cells, boundary_condition, neighbour_order)
+
+    if isinstance(hopping, (int, float, complex)):
+        hopping = [hopping]
+
+    hopping = math.asarray(hopping)
+
+    if isinstance(intersite_int, (int, float, complex)):
+        intersite_int = [intersite_int]
+
+    intersite_int = math.asarray(intersite_int)
+
+    if hopping.shape not in [(neighbour_order,), (lattice.n_sites, lattice.n_sites)]:
+        raise ValueError(
+            f"The hopping parameter should be a number or an array of shape ({neighbour_order},) or ({lattice.n_sites},{lattice.n_sites})"
+        )
+
+    if intersite_int.shape not in [(neighbour_order,), (lattice.n_sites, lattice.n_sites)]:
+        raise ValueError(
+            f"The intersite_int parameter should be a number or an array of shape ({neighbour_order},) or ({lattice.n_sites},{lattice.n_sites})"
+        )
+
+    spin = 2
+    hopping_ham = 0.0 * FermiWord({})
+    intersite_term = 0.0 * FermiWord({})
+    if hopping.shape == (neighbour_order,):
```

Comment:
How do these `if`-`else` cases differ other than accessing `hopping`? Maybe we could rather restrict conditional to it to prevent repeating code whenever possible.